handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -432,7 +435,13 @@ public final SSLEngine newEngine(ByteBufAllocator alloc) {
      */
     @Deprecated
     public final long context() {
-        return ctx;
+        Lock lock = ctxLock.readLock();
```

Comment:
also consider adding a comment to the javadocs that care must be taken with this pointer as it may be freed.